WT vs *MZsap130a* 36hpf
================
Ricardo A. DeMoya
2023-06-18

# RNAseq analysis pipeline

The csv file that contains the count data to b uploaded, is from total
RNA collected from whole embryos at 36hpf for WT and *MZsap130a*
genotypes. The *MZsap130a* animals were further separated into the
normal ventricle (NV) or small ventricle (SV) groups for RNA extraction.
The total RNA was sent to sequencing core at the University of
Pittsburgh with three pooled biological replicates for each group
(WT1-3, NV1-3, and SV1-3, ~75 embryos per sample). Once the *fastq*
files were obtain we uploaded them in CLC Genomics Workbench by QIAGEN
and used the RNAseq analysis tool to produce a Total count matrix for
export as a csv file. This file was then used in R and Rstudio to
explore the relationship between WT and *MZsap130a* mutants.

## Data input and metadata

Using the read.table() function, we import the data table from a csv
count matrix generated by CLC Gx workbench. Other alignment methods can
be used including, Rbowtie2, TopHat2, Rsubread, or HISAT2. Annotation
was done using GRCz11 zebrafish reference genome.

### Read in the expression count matrix, exported from CLC Gx after annotation is done with GRCz11

``` r
data_edger = read.table(
  file="36hpf_RNAseq_sap130a_pt32a_vs_AB_count_matrix_2.csv",
  sep=",",
  header=T)
head(data_edger[,1:4], n = 5)
```

    ##            Feature.ID        ENSEMBL.IDs AB_1_36hpf AB_3_36hpf
    ## 1                a1cf ENSDARG00000002968        295        361
    ## 2                a2ml ENSDARG00000056314       6828       8808
    ## 3 A2ML1 (1 of many)_1 ENSDARG00000115205         14          9
    ## 4 A2ML1 (1 of many)_2 ENSDARG00000008835          1          4
    ## 5 A2ML1 (1 of many)_3 ENSDARG00000041645       2227       2480

### Updating gene names from GRCz11

``` r
data_edger$Feature.ID[data_edger$Feature.ID == "ACTC1"] <- "actc1"
data_edger$Feature.ID[data_edger$Feature.ID == "FP102018.1"] <- "ltbp3"
```

### Making a data frame and setting the rownames to genenames or ENSEMBL IDs

``` r
data_edger <- data.frame(data_edger, row.names = data_edger$Feature.ID)
```

### Double checking for duplicates

``` r
# Makes a list of ENSEMBL IDs in the data set and makes sure there are no duplicates
unique_edger = names(which(table(as.vector(data_edger[,2])) == 1))
# Indexes your data for rows that are present in the list you made in the last step
unique.data_edger = data_edger[which(data_edger[,2] %in% unique_edger),]

# Repackage as a dataframe
unique.data_edger = data.frame(unique.data_edger)
# This makes the rownames equal to genenames
rownames(unique.data_edger) = unique.data_edger[,1] # change to 2 for IDs
# This will remove the ID and names columns in this data set column 1 and 2
unique.data_edger = unique.data_edger[,-1:-2]
```

### Creating a metadata table

``` r
# Creates a design matrix for modeling the sequencing experiment and any co-variants
# WT, NV, SV all separate
coldat <- data.frame(row.names = colnames(unique.data_edger), stringsAsFactors = T)
coldat$Groups <- factor(c(rep("WT", 3), rep("NV", 3), rep("SV",3)),
                        levels = c("WT","NV","SV"))
coldat$Genotype <- factor(c(rep("AB*",3),rep("sap130a^pt32a/pt32a", 6)))
coldat$Colors <- c(rep('blue',3), rep('red',3),rep('black',3))
coldat$Shapes <- c(rep(19,3),rep(15,3),rep(17,3))

# Check out the metadata table
coldat
```

    ##                   Groups            Genotype Colors Shapes
    ## AB_1_36hpf            WT                 AB*   blue     19
    ## AB_3_36hpf            WT                 AB*   blue     19
    ## AB_2_36hpf            WT                 AB*   blue     19
    ## sap130aNV_3_36hpf     NV sap130a^pt32a/pt32a    red     15
    ## sap130aNV_2_36hpf     NV sap130a^pt32a/pt32a    red     15
    ## sap130aNV_1_36hpf     NV sap130a^pt32a/pt32a    red     15
    ## sap130aSV_2_36hpf     SV sap130a^pt32a/pt32a  black     17
    ## sap130aSV_1_36hpf     SV sap130a^pt32a/pt32a  black     17
    ## sap130aSV_3_36hpf     SV sap130a^pt32a/pt32a  black     17

## Data exploration before RNAseq analysis

### MDS plot before normalization

The multidimensional scaling (MDS) plot, uses a PCA plot constructed
from the top genes with largest standard deviations across the samples.
These data should provide a relationship among samples, how they cluster
together or separate.

``` r
# Make sure you have loaded the necessary libraries for the analysis 
suppressMessages(library('edgeR'))
library('limma')

# MDS to visualizes the differences between the expression profiles of different
# samples in two dimensions
mds_edger <- plotMDS(x = unique.data_edger,gene.selection = "common", 
                     top = nrow(unique.data_edger),
                     col = coldat$Colors, pch = coldat$Shapes,
                     cex = 2.5,
                     main = "Multidimensional Scaling Plot\nbefore normalization")
grid(lty = 6)
legend('bottomleft',legend = c('WT','NV','SV'),
       col =c('blue','red','black'),pch = c(19,15,17),cex = 1)
```

![](MZsap130a_vs_AB_36hpf_wholeEmbryo_files/figure-gfm/MDS%20before-1.png)<!-- -->

Euclidean distances on the plot approximate the typical log2 fold
changes between the samples. Distances on the plot can be interpreted as
leading log2-fold-change, meaning the typical (root-mean-square)
log2-fold-change between the samples for the genes that distinguish
those samples

### Making a DGEList data object for edgeR RNAseq analysis

A simple list-based data object used in the edgeR package

``` r
# Create DGEList object for normalization and differential gene expression analysis
y_edgeR = DGEList(counts=unique.data_edger,
                  genes=row.names(unique.data_edger),
                  samples = coldat,
                  group = coldat$Groups)
```

### Filter lowly expressed genes

``` r
# Default removes all gene rows with less than 15 transcripts across any three
# samples (min.total.count=15) and keeps gene row with 10+ transcripts in more
# than 70% across samples (min.count=10,min.prop=0.7)
keep <- filterByExpr(y_edgeR,group = y_edgeR$samples$Groups)
y_edgeR <- y_edgeR[keep, , keep.lib.sizes=F]
```

### Normalize the data

``` r
y_edgeR <- calcNormFactors(y_edgeR, method = "TMM") # Trimmed Mean of M-values (TMM)
```

## What influence did the normalization have on the data

We want to see the samples cluster in their own groups, revealing
differences between WT and the *MZsap130a* mutants

``` r
library("ggplot2")
# Another MDS plot to look at sample grouping
mds_edger2 <- plotMDS(y_edgeR,gene.selection = "common",
                      top = nrow(y_edgeR),
                      col = coldat$Colors,
                      pch = coldat$Shapes,
                      main = "Multidimensional Scaling Plot\nafter normalization",
                      cex = 2.5)+
  theme(axis.title = element_text(size = 30, face = "bold"))
grid(lty = 6)
legend('topleft',legend = c('WT','NV','SV'),
       col =c('blue','red','black'),pch = c(19,15,17),cex = 1)
```

![](MZsap130a_vs_AB_36hpf_wholeEmbryo_files/figure-gfm/MDS%20after-1.png)<!-- -->

## Plotting a mean-difference graph to see if the TMM normalization removed composition bias from the libraries

It is important to check the TMM normalization for expected results. So
here we look at one sample from each group.

``` r
# TMM normalization should have the bulk of genes centered at 0, meaning the
# composition bias from the library sizes has been successfully removed
AB <- edgeR::plotMD.DGEList(y_edgeR, column = 2)+ # Change the sample being plotted
# by changing column = 1-9, for samples 1-9
abline(h=0, col="red", lty=2, lwd=2)
```

![](MZsap130a_vs_AB_36hpf_wholeEmbryo_files/figure-gfm/Normalization%20check%20WT-1.png)<!-- -->

``` r
MZsap130a <- edgeR::plotMD.DGEList(y_edgeR, column = 6)+
abline(h=0, col="red", lty=2, lwd=2)
```

![](MZsap130a_vs_AB_36hpf_wholeEmbryo_files/figure-gfm/Normalization%20check%20NV-1.png)<!-- -->

``` r
MZsap130a <- edgeR::plotMD.DGEList(y_edgeR, column = 9)+
abline(h=0, col="red", lty=2, lwd=2)
```

![](MZsap130a_vs_AB_36hpf_wholeEmbryo_files/figure-gfm/Normalization%20check%20SV-1.png)<!-- -->

All groups are centered around zero as expected, so the TMM
normalization worked well to removed composition bias from the libraries

### Creating design matricies to compare WT vs NV vs SV or WT vs *MZsap130a*

``` r
# Next is the design matrix used to provide info on the samples, in this case the
# groups of samples collected WT, NV, and SV
design_WTvsNVvsSV <- model.matrix(~coldat$Groups)
rownames(design_WTvsNVvsSV) <- rownames(coldat)
colnames(design_WTvsNVvsSV) <- c("WT","NV","SV")

# This design matrix will compare the genotypes
design_WTvsMZsap130a <- model.matrix(~coldat$Genotype)
rownames(design_WTvsMZsap130a) <- rownames(coldat)
colnames(design_WTvsMZsap130a) <- c("WT","MZsap130a")

# The WT data is always the intercept for comparison to NV or SV groups and all
# MZsap130a samples
design_WTvsNVvsSV
```

    ##                   WT NV SV
    ## AB_1_36hpf         1  0  0
    ## AB_3_36hpf         1  0  0
    ## AB_2_36hpf         1  0  0
    ## sap130aNV_3_36hpf  1  1  0
    ## sap130aNV_2_36hpf  1  1  0
    ## sap130aNV_1_36hpf  1  1  0
    ## sap130aSV_2_36hpf  1  0  1
    ## sap130aSV_1_36hpf  1  0  1
    ## sap130aSV_3_36hpf  1  0  1
    ## attr(,"assign")
    ## [1] 0 1 1
    ## attr(,"contrasts")
    ## attr(,"contrasts")$`coldat$Groups`
    ## [1] "contr.treatment"

``` r
design_WTvsMZsap130a
```

    ##                   WT MZsap130a
    ## AB_1_36hpf         1         0
    ## AB_3_36hpf         1         0
    ## AB_2_36hpf         1         0
    ## sap130aNV_3_36hpf  1         1
    ## sap130aNV_2_36hpf  1         1
    ## sap130aNV_1_36hpf  1         1
    ## sap130aSV_2_36hpf  1         1
    ## sap130aSV_1_36hpf  1         1
    ## sap130aSV_3_36hpf  1         1
    ## attr(,"assign")
    ## [1] 0 1
    ## attr(,"contrasts")
    ## attr(,"contrasts")$`coldat$Genotype`
    ## [1] "contr.treatment"

Here I wanted to search for differences between the NV and SV groups
when compared to WT, to accomplish this edgeR implements an ANOVA-like
analysis across the samples. I hoped to find differences that could
explain the difference in phenotype seen among the *MZsap130a* mutants.

### Estimate the dispersions among genes and samples comparing WTvsNVvsSV

``` r
# Estimate dispersions
# Common dispersion = the mean dispersion across all genes
y_edger = estimateGLMCommonDisp(y_edgeR,design_WTvsNVvsSV,verbose = T)
```

    ## Disp = 0.0215 , BCV = 0.1466

``` r
# Trended dispersion = the mean dispersion across all genes with similar abundances
# In other words, the fitted value of the mean-dispersion trend.
y_edger = estimateGLMTrendedDisp(y_edger,design_WTvsNVvsSV)

# Tagwise dispersion = the gene specific dispersions, so across samples
y_edger = estimateGLMTagwiseDisp(y_edger,design_WTvsNVvsSV)
```

### Plotting the biological coefficient of variation for WTvsNVvsSV

This allows for visualization of the biological variance you wish to
study. It represents the covariate of variation that would remain
between biological replicates if sequencing depth could be increased
indefinitely. For human cancer RNAseq data a BCV around 0.40 is common
but with isogeneic mouse lines this drops to 0.10, anything between
these values is a reasonable dataset to explore further.

``` r
# Make a variable to hold the Biological coefficient of variation (BCV)
bcv <- sqrt(as.numeric(y_edger[["common.dispersion"]]))

# plots the tagwise biological coefficient of variation (square root of dispersions)
# against log2-CPM 
plotBCV(y_edger, xlab = "Average Log2CPM")
text(x = 11,y =0.4, labels = round(bcv, digits = 5))
```

![](MZsap130a_vs_AB_36hpf_wholeEmbryo_files/figure-gfm/BCV%20plot%20WTvsNVvsSV-1.png)<!-- -->

### Estimate the dispersions among genes and samples comparing WT vs *MZsap130a*

``` r
# Estimate dispersions
# Common dispersion = the mean dispersion across all genes
y_edgerMZ = estimateGLMCommonDisp(y_edgeR,design_WTvsMZsap130a,verbose = T)
```

    ## Disp = 0.0233 , BCV = 0.1526

``` r
# Trended dispersion = the mean dispersion across all genes with similar abundances. 
# In other words, the fitted value of the mean-dispersion trend.
y_edgerMZ = estimateGLMTrendedDisp(y_edgerMZ,design_WTvsMZsap130a)

# Tagwise dispersion = the gene specific dispersions, across samples
y_edgerMZ = estimateGLMTagwiseDisp(y_edgerMZ,design_WTvsMZsap130a)
```

### Plotting the biological coefficient of variation for WT vs *MZsap130a*

Noticed a slight increase in BCV, suggesting combining the NV and SV
groups into one MZsap130a group, reveals more differences from WTs

``` r
# Make a variable to hold the Biological coefficient of variation (BCV)
bcv <- sqrt(as.numeric(y_edgerMZ[["common.dispersion"]]))

# The tagwise biological coefficient of variation (square root of common dispersion)
# against log2-CPM 
plotBCV(y_edgerMZ, xlab = "Average Log2CPM")
text(x = 11,y =0.4, labels = round(bcv, digits = 5))
```

![](MZsap130a_vs_AB_36hpf_wholeEmbryo_files/figure-gfm/BCV%20plot%20WTvsMZsap130a-1.png)<!-- -->

### Now we fit the model using glmQLFit()

``` r
# Fit to the model comparing WTvsNVvsSV
fit_edger = edgeR::glmQLFit(y_edger, design = design_WTvsNVvsSV)

# Fit to the model comparing WTvsMZsap130a
fit_edgerMZ = edgeR::glmQLFit(y_edger, design = design_WTvsMZsap130a)
```

glmQLFit() uses the trended NB dispersion to fit a Generalized Linear
Model (GLM), and then estimates the Quasi-Likelihood (QL) dispersion
from the deviance. It doesn’t re-estimate the NB dispersion but instead
estimates an entirely different QL dispersion for each gene. In
contrast, glmFit() uses the tagwise NB dispersion to fit a GLM - and
that’s it.

### Finally the analysis using glmQFLTest() and topTags() on the fitted data from the previous step

``` r
# The analysis for WT vs NV vs SV, ANOVA-like edgeR
edgeR.results_ql = as.data.frame(edgeR::topTags(
  edgeR::glmQLFTest(glmfit = fit_edger, coef = c(2:3)),n=nrow(unique.data_edger)))

# The analysis for WT vs MZsap130a, ANOVA-like edgeR
edgeR.results_qlMZ = as.data.frame(edgeR::topTags(
  edgeR::glmQLFTest(glmfit = fit_edgerMZ),n=nrow(unique.data_edger)))
#write.csv(edgeR.results_qlMZ, file = "MZsap130_AB_ALL_DEGs.csv")
```

### Do *MZsap130a* NV and SV groups have transcript differences that can explain their physical differences

Here I looked at the genes that had opposite trends in the NVs and the
SVs. This revealed some genes that have not been identified or studied
in zebrafish heart development.

``` r
## Upregulated genes from AB vs MZsap130a
upreg <- edgeR.results_qlMZ[which(edgeR.results_qlMZ$logFC>2 & edgeR.results_qlMZ$FDR<0.05),]
#write.csv(upreg,file = "MZsap130_AB_2logFC_fdr0.05.csv")
# Keeps significant genes with FDR<=0.05 and 
#abs(log2FC) >= 0.4 (which = 1.3+ or - fold change)
Sig_edgeR_DEGs <- edgeR.results_ql[
  which(edgeR.results_ql$FDR<=0.05 & abs(edgeR.results_ql$logFC.SV)>=0.4),]

Sig_edgeR_DEG_NV <- edgeR.results_ql[
  which(edgeR.results_ql$FDR<=0.05 & abs(edgeR.results_ql$logFC.NV)>=0.4),]

# Genes that are significant in WT vs NV and not WT vs SV
sig_inNV_notSV <- Sig_edgeR_DEG_NV[
  -(which(Sig_edgeR_DEG_NV$genes %in% Sig_edgeR_DEGs$genes)),]

# Keep genes with a difference in log2FC of 0.6
sig_diff_NV <- sig_inNV_notSV[
  which((sig_inNV_notSV$logFC.NV-sig_inNV_notSV$logFC.SV)>=0.6),]

# Genes that are significant in WT vs SV and not WT vs NV
sig_inSV_notNV <- Sig_edgeR_DEGs[
  -(which(Sig_edgeR_DEG_NV$genes %in% Sig_edgeR_DEGs$genes)),]

sig_diff_SV <- sig_inSV_notNV[
  which((sig_inSV_notNV$logFC.NV-sig_inSV_notNV$logFC.SV)>=0.6),]

dim(sig_diff_NV) # Only six genes showing large 0.6 difference
```

    ## [1] 7 7

``` r
dim(sig_diff_SV) # 22 genes or rows
```

    ## [1] 25  7

# Basic RNAseq Plots to Explore the Data

## Volcano plots

### Libraries needed for volcano plots

``` r
# Libraries for plotting data
library('ggplot2')

# Load packages
suppressMessages(library('tidyverse'))
library('ggrepel')
```

### Data preparation for volcano plots

``` r
volcano_datSV <- data.frame (GeneNames = rownames(edgeR.results_ql),
                             log2FC = edgeR.results_ql$logFC.SV,
                             FDR = edgeR.results_ql$FDR)

rownames(volcano_datSV) <- volcano_datSV$GeneNames

# making a column that labels who is in which group for plotting
volcano_datSV <- volcano_datSV %>% 
  mutate(
    Significance = case_when(
      abs(log2FC) >= 0.4 & FDR < 0.05 ~ "log2FC & FDR", 
      abs(log2FC) <= 0.4 & FDR < 0.05 ~ "FDR",
      abs(log2FC) >= 0.4 & FDR > 0.05 ~ "log2FC", 
      abs(log2FC) <= 0.4 & FDR > 0.05 ~ "NS")
  )
```

### Lets make another for WT vs NV

``` r
volcano_datNV <- data.frame (GeneNames = rownames(edgeR.results_ql),
                             log2FC = edgeR.results_ql$logFC.NV,
                             FDR = edgeR.results_ql$FDR)

rownames(volcano_datNV) <- volcano_datNV$GeneNames

#making a column that labels who is in which group for plotting
volcano_datNV <- volcano_datNV %>% 
  mutate(
    Significance = case_when(
      abs(log2FC) >= 0.4 & FDR < 0.05 ~ "log2FC & FDR", 
      abs(log2FC) <= 0.4 & FDR < 0.05 ~ "FDR",
      abs(log2FC) >= 0.4 & FDR > 0.05 ~ "log2FC", 
      abs(log2FC) <= 0.4 & FDR > 0.05 ~ "NS")
  )
```

### Selected genes for labels in the volcano plot

``` r
# Selected genenames for volcano plots
vol_cellComm_Sarcomere <- c("jph2","jph1b","sgca","cxcr4b",
                            "actc1","ttn.1","ttn.2","acta1a")
                            
vol_sarco_cellCycle <-c("actc1","ttn.1","ttn.2",
                        "e2f7","brca2","vrk1","haus6")

vol_Mito_metabo <- c("mtrf1","mrpl58","slc25a3a",
                     "acox","acsl1a", #fatty acid
                     "ugpa2a","phkb") #glycogen metabo.

vol_OFT_SmoMuscle <- c("isl1",
                       "tagln2","cnn2")
```

### Overlaping genes found from DEG analyis of *sap130b* vs WT and *sap130a* vs WT at 36hpf whole embryos

``` r
# List for the overlap DEGs in MZsap130a vs MZsap130b at 36hpf whole embryo
overlap <- c("hbbe2","hbbe3","hebp2","hbae3","hbae5", # 1st row hemoglobins
             "cyp4v2a","cyp1b1","cyp46a1.1","cyp2x6") # 2nd row is cytochromes
# associated with heme, iron, oxygen binding

sarco <- c("actc1","ttn.1","ttn.2","myh6","myl4","cacng1a","myh7","tpm4a","tcap","tnnt2a")

newnew <- c("actc1","cxcr4b","myh6","myh7","myl7","myzap","ttn.1","gja3","myl4")
```

### Set up the theme for our volcano plot, very powerful tool for consistant plotting

``` r
# Theme set up to control all elements
mi_teme <- theme(title = element_text(size = 14, face = "bold"),
                 plot.title.position = "panel",
                 plot.title = element_text(vjust =-2,hjust = 0.5),
                 axis.text.x = element_text(size= 10),
                 axis.text.y = element_text(size = 10),
                 axis.title.x = element_text(face = "bold",size = 12),
                 axis.title.y = element_text(face = "bold",size = 12),
                 legend.text = element_text(size = 10, face = "bold"),
                 legend.title = element_blank(),
                 legend.position = "top",
                 legend.spacing.x = unit(0.2,"picas"), 
                 legend.key = element_blank(),
                 legend.box.spacing = unit(0.001,"picas"),
                 legend.key.size = unit(0.001,"picas"),
                 plot.margin = unit(c(0.01,1,0.01,1),"mm"))
```

### Make a volcano plot to visualize all the data at once

Here I plot all the genes in one graph with cuts offs set to an FDR
\<0.05 and a log2FC \> 0.4 (~1.3logFC). Looking for heart specific
changes with whole embryo resolution we wanted to include lower log2FC
values, to increase the odds of finding changes we know are important to
heart development.

``` r
# Make a list of desired genes to plot
s <- volcano_datSV %>%
  filter(GeneNames %in% vol_Mito_metabo)

# Make a ggplot volcano, similar to EnhancedVolcano
# title etc
volcanoMan <- ggplot(volcano_datSV, aes(log2FC,-log(FDR,10)))+
  geom_point(aes(color=Significance),size = 0.5, alpha = 4/5)+
  scale_color_manual(values = c("NS"="darkgrey","log2FC & FDR"="seagreen",
                                "log2FC"="orangered3","FDR"="lightblue"))+
  xlab(bquote(~Log[2]~fold~change))+ xlim(-5,5)+
  ylab(bquote(~-Log[10]~italic(FDR)))+ylim(0,6)+
  geom_hline(yintercept = -log10(0.05))+
  geom_vline(xintercept = c(-0.4,0.4))+
  ggtitle(expression(bolditalic("Tg(myl7:EGFP)")~bold("vs")~
                       bolditalic("MZsap130a;Tg(myl7:EGFP)"~bold("SV"))))+
  geom_label_repel(data = s,mapping = aes(label = GeneNames), size = 4, 
                  force = 0.5,direction = "both",
                  nudge_y = c(rep(0.35,5),rep(-.25,2),rep(0.35,3)),
                  arrow = arrow(length = unit(0.01, 'npc'),ends = "last",type = "closed"),
                  segment.color = 'black',segment.size = 0.4,max.iter = 3e3,
                  min.segment.length = 0, seed = 42,
                  fontface = "bold.italic",label.padding = 0.1)+
  geom_point(data =s ,color = "red",size=1)+
  guides(color= guide_legend(reverse = TRUE, override.aes = list(size=5)))

suppressWarnings(print(volcanoMan + mi_teme))
```

![](MZsap130a_vs_AB_36hpf_wholeEmbryo_files/figure-gfm/WT%20vs%20SV%2036hpf%20whole%20embryo-1.png)<!-- -->

### Taking a look at the WT vs NV data

``` r
# Make a list of desired genes to plot
s <- volcano_datNV %>%
  filter(GeneNames %in% vol_Mito_metabo)

# Make a ggplot volcano, similar to EnhancedVolcano
# title etc
volcanoMan <- ggplot(volcano_datNV, aes(log2FC,-log(FDR,10)))+
  geom_point(aes(color=Significance),size = 0.5, alpha = 4/5)+
  scale_color_manual(values = c("NS"="darkgrey","log2FC & FDR"="seagreen",
                                "log2FC"="orangered3","FDR"="lightblue"))+
  xlab(bquote(~Log[2]~fold~change))+ xlim(-5,5)+
  ylab(bquote(~-Log[10]~italic(FDR)))+ylim(0,6)+
  geom_hline(yintercept = -log10(0.05))+
  geom_vline(xintercept = c(-0.4,0.4))+
  ggtitle(expression(bolditalic("Tg(myl7:EGFP)")~bold("vs")~
                       bolditalic("MZsap130a;Tg(myl7:EGFP)"~bold("NV"))))+
  geom_label_repel(data = s,mapping = aes(label = GeneNames), size = 4, 
                  force = 0.5,direction = "both",
                  nudge_y = c(rep(0.35,5),rep(-.25,2),rep(0.35,3)),
                  arrow = arrow(length = unit(0.01, 'npc'),ends = "last",type = "closed"),
                  segment.color = 'black',segment.size = 0.4,max.iter = 3e3,
                  min.segment.length = 0, seed = 42,
                  fontface = "bold.italic",label.padding = 0.1)+
  geom_point(data =s ,color = "red",size=0.6)+
  guides(color= guide_legend(reverse = TRUE, override.aes = list(size=2.5)))

suppressWarnings(print(volcanoMan + mi_teme))
```

![](MZsap130a_vs_AB_36hpf_wholeEmbryo_files/figure-gfm/WT%20vs%20NV%2036hpf%20whole%20embryo-1.png)<!-- -->

# Summary

This report summarizes the basic pipeline used to analyze the
*MZsap130a* and *MZsap130b* whole embryo RNAseq data sets. All plots for
the publication were made using ggplot2, heatmap3, viridisLite,
reshape2, dendextend, tidyverse, and ggrepel. The data is explored and
analyzed here based on the edgeR user’s guide found here at
[Bioconductor](https://bioconductor.org/packages/release/bioc/html/edgeR.html)
